bld/namelist_files/namelist_defaults_ctsm.xml

```diff
@@ -2586,4 +2586,14 @@ lnd/clm2/surfdata_map/release-clm5.0.30/surfdata_ne0np4.CONUS.ne30x8_hist_78pfts
 
 <init_interp_method>general</init_interp_method>
 
+<!-- =========================================  -->
+<!-- Defaults for excess_ice                    -->
+<!-- =========================================  -->
+
+<use_excess_ice>.false.</use_excess_ice>
+
+<stream_fldfilename_exice use_excess_ice=".true.">lnd/clm2/paramdata/exice_init_0.125x0.125_c20220516.nc</stream_fldfilename_exice>
```

Comment:
This isn't on cheyenne yet. We should get a copy to look at. Could you copy it to somewhere on cheyenne and let us know where it's at so we can look at it?

You also may not have permission to put under /glade/p/cesmdata/cseg/inputdata on cheyenne. We should discuss if you given permission to do so. If you won't be doing this often we won't need to. If you are going to be doing so, we'll need to do some training with you before we give you the permissions.

Reply:
I'm also not sure if the paramdata subdirectory is the right location for this, but we need to look at it to make that assessment.

Reply:
In the initial comment, there is [a google drive](https://drive.google.com/file/d/1mA457Oa52zG_MtLGB7KHuUYQvsS2-P5o/view?usp=sharing) link to the zipped file that contains both the stream file and the mesh file. I do not have access to cheyenne or glade. I don't think I will need one. I will ask Kaveh to put it there or if you can put it there, it would be awesome.

Regarding paramdata subdirectory -- I just followed the way ch4finundated is setup i.e. namelist defaults for _finundated_inversiondata*_ files are located.

> The last more detailed thing is how this will be used in practice regarding restarts and such. The stream file gives the level of excess ice at some point in time. 

Since there is only one dataset with initial concentration for excess ice concentration (unitless, for the part of the soil column, which is allowed to have excess ice in it) -- I followed the way _finundated inversion data_ is set up.

> If you want to run for a different date (especially a future date) the ice levels should have melted. And as such you should startup from levels from a restart file. Does that mean if you do a startup-type run, it'll use values from the stream file, but if you do a hybrid or branch it'll be from the restart file and the stream is ignored? 

Yes. This is correct, in the _WaterStateType%Restart_ if the variable is not found on the restart file or file containing initial conditions datasets -- the stream data will be used. There is no way this can work for branch-type runs since they do not allow to have missing variables in the restart file.  

This is done mainly for backward compatibility, so runs with `use_excess_ice==.true.` can be started from initial conditions that already exist (even if they don't have excess ice). In practice, it means that users should run a small spin-up run (I am looking into how long that should typically take) when they want to run excess ice option, for the soil hydro-thermal/bgc state to equilibrate.

For initial conditions and restart files that will be created with `use_excess_ice==.true.`, since the variable exists in the files it will be read and the stream value won't be used. However, the stream data is still read in any run with `use_excess_ice==.true.`. 
I was looking into the possibility of adding a logical variable that will look at if the `use_excess_ice==.true. has been already run, so there is no need to read stream data at all, but I am not sure that a small initialization cap/file download is worth overcomplicating things. Maybe @dlawrenncar and @swensosc  have better judgment for that.

Reply:
OK, I grabbed those files and have put them on cheyenne and also on our inputdata repository.  I had to change the format of the Mesh file from NetCDF-4 to CDF5. I'm still not sure if paramdata is the right subdirectory for this, but since the methane data is also there, it's at least consistent.

Reply:
@mvdebolskiy can you give me push access to your fork, so I can push the change about the file? You do that by going to settings on github in your fork and then collaborators.

Reply:
Done, check for an invite

Reply:
OK I pushed my changes.